### PR TITLE
add keras2 GlobalAveragePooling2D  and GlobaMaxPooling1D  layers 

### PR DESCRIPTION
**Pull request by yjjiang11**

### CODE REVIEW

**Comment by yjjiang11** [comment]
Jenkens passed: http://172.168.2.101:8080/view/ZOO-PR/job/ZOO-PR-Validation/733/ @hkvision

File: pyzoo/zoo/pipeline/api/keras2/layers/pooling.py [hunk 1]
```diff
@@ -121,3 +121,31 @@ def __init__(self,
...
+class GlobalAveragePooling2D(ZooKeras2Layer):
+    """
+    Applies global average pooling operation for spatial data.
+    The input of this layer should be 4D.
+
+    When you use this layer as the first layer of a model, you need to provide the argument
+    input_shape (a shape tuple, does not include the batch dimension).
+
+    # Arguments
+    padding: Format of input data. Either 'th' (Channel First) or 'tf' (Channel Last).
```
Comment (hkvision) [inline pyzoo/zoo/pipeline/api/keras2/layers/pooling.py]: Is it the same with Keras? should be data_format.... https://keras.io/layers/pooling/#globalaveragepooling2d

File: zoo/src/main/scala/com/intel/analytics/zoo/pipeline/api/keras2/layers/GlobalAveragePooling2D.scala [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+ * When you use this layer as the first layer of a model, you need to provide the argument
+ * inputShape (a Single Shape, does not include the batch dimension).
+ *
+ * @param dimOrdering Format of input data. Please use DataFormat.NCHW (dimOrdering='th')
+ *                    or DataFormat.NHWC (dimOrdering='tf'). Default is NCHW.
+ * @param inputShape A Single Shape, does not include the batch dimension.
+ * @tparam T The numeric type of parameter(e.g. weight, bias). Only support float/double now.
+ */
+class GlobalAveragePooling2D[T: ClassTag](
+      override val dimOrdering: DataFormat = DataFormat.NCHW,
```
Comment (hkvision) [inline zoo/src/main/scala/com/intel/analytics/zoo/pipeline/api/keras2/layers/GlobalAveragePooling2D.scala]: dataFormat

File: zoo/src/main/scala/com/intel/analytics/zoo/pipeline/api/keras2/layers/GlobalAveragePooling2D.scala [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+      override val inputShape: Shape = null)(implicit ev: TensorNumeric[T])
+  extends klayers1.GlobalAveragePooling2D[T](
+    dimOrdering, inputShape) with Net {}
+
+object GlobalAveragePooling2D {
+  def apply[@specialized(Float, Double) T: ClassTag](
+      dimOrdering: String = "th",
```
Comment (hkvision) [inline zoo/src/main/scala/com/intel/analytics/zoo/pipeline/api/keras2/layers/GlobalAveragePooling2D.scala]: channel_first, channel_last. See keras doc for more details.

File: zoo/src/test/scala/com/intel/analytics/zoo/pipeline/api/keras2/layers/GlobalAveragePooling2DSpec.scala [hunk 4]
```diff
@@ -48,11 +48,11 @@ class GlobalAveragePooling2DSpec extends KerasBaseSpec {
       """
         |input_tensor = Input(shape=[32, 28, 6])
         |input = np.random.random([3, 32, 28, 6])
-        |output_tensor = GlobalAveragePooling2D(dim_ordering="tf")(input_tensor)
+        |output_tensor = GlobalAveragePooling2D(dataFormat="th")(input_tensor)
```
Comment (hkvision) [inline zoo/src/test/scala/com/intel/analytics/zoo/pipeline/api/keras2/layers/GlobalAveragePooling2DSpec.scala]: I think in Keras2, data format uses "channel_first"/"channel_last"

**Comment by yjjiang11** [comment]
http://172.168.2.101:8080/view/ZOO-PR/job/ZOO-PR-Validation/743/

**Comment by yjjiang11** [comment]
http://172.168.2.101:8080/view/ZOO-PR/job/ZOO-PR-Validation/743/

File: zoo/src/main/scala/com/intel/analytics/zoo/pipeline/api/keras2/layers/utils/KerasUtils.scala [hunk 5]
```diff
@@ -0,0 +1,42 @@
...
+package com.intel.analytics.zoo.pipeline.api.keras2.layers.utils
+
+
+import com.intel.analytics.bigdl.nn.abstractnn.DataFormat
+
+
+object KerasUtils {
+
+  def toBigDLFormat(dataFormat: String): DataFormat = {
+    require(dataFormat.toLowerCase() == "channels_first",
```
Comment (zhichao-li) [inline zoo/src/main/scala/com/intel/analytics/zoo/pipeline/api/keras2/layers/utils/KerasUtils.scala]: channels_first => NCHW; channels_last => NHWC ?

**Comment by yjjiang11** [comment]
Jenkins passed:http://172.168.2.101:8080/view/ZOO-PR/job/ZOO-PR-Validation/745/ @hkvision

File: pyzoo/zoo/pipeline/api/keras2/layers/pooling.py [hunk 6]
```diff
@@ -121,3 +121,54 @@ def __init__(self,
...
+class GlobalAveragePooling2D(ZooKeras2Layer):
+    """
+    Applies global average pooling operation for spatial data.
+    The input of this layer should be 4D.
+
+    When you use this layer as the first layer of a model, you need to provide the argument
+    input_shape (a shape tuple, does not include the batch dimension).
+
+    # Arguments
+    data_format: Format of input data. Channel First .
```
Comment (hkvision) [inline pyzoo/zoo/pipeline/api/keras2/layers/pooling.py]: ?? Either "channel_first" or "channel_last"

File: zoo/src/main/scala/com/intel/analytics/zoo/pipeline/api/keras2/layers/utils/KerasUtils.scala [hunk 7]
```diff
@@ -0,0 +1,48 @@
...
+
+import com.intel.analytics.bigdl.nn.abstractnn.DataFormat
+
+
+object KerasUtils {
+
+  def toBigDLFormat(dataFormat: String): DataFormat = {
+    require(dataFormat.toLowerCase() == "channels_" +
+      "first" || dataFormat.toLowerCase() == "channels_last",
+      s"Dim ordering must be either channels_first or " +
```
Comment (hkvision) [inline zoo/src/main/scala/com/intel/analytics/zoo/pipeline/api/keras2/layers/utils/KerasUtils.scala]: Change Dim ordering to Data format

File: zoo/src/main/scala/com/intel/analytics/zoo/pipeline/api/keras2/layers/utils/KerasUtils.scala [hunk 8]
```diff
@@ -0,0 +1,48 @@
...
+    dataFormat.toLowerCase() match {
+      case "channels_last" => DataFormat.NHWC
+      case "channels_first" => DataFormat.NCHW
+    }
+  }
+
+  def toBigDLFormat5D(dataFormat: String): String = {
+    require(dataFormat.toLowerCase() == "channels_fir" +
+      "st" || dataFormat.toLowerCase() == "channels_last",
+      s"Dim ordering must be either channels_first or" +
```
Comment (hkvision) [inline zoo/src/main/scala/com/intel/analytics/zoo/pipeline/api/keras2/layers/utils/KerasUtils.scala]: same as above

File: zoo/src/test/scala/com/intel/analytics/zoo/pipeline/api/keras2/layers/GlobalAveragePooling2DSpec.scala [hunk 9]
```diff
@@ -0,0 +1,71 @@
...
+import scala.util.Random
+
+class GlobalAveragePooling2DSpec extends KerasBaseSpec {
+
+  "GlobalAveragePooling2D NCHW" should "be the same as Keras" taggedAs(Keras2Test) in {
+    val kerasCode =
+      """
+        |input_tensor = Input(shape=[3, 16, 20])
+        |input = np.random.random([2, 3, 16, 20])
+        |output_tensor = GlobalAveragePooling2D(dim_ordering="channels_first")(input_tensor)
```
Comment (hkvision) [inline zoo/src/test/scala/com/intel/analytics/zoo/pipeline/api/keras2/layers/GlobalAveragePooling2DSpec.scala]: shouldn't it be "data_format"?
jenkins failed http://10.239.47.211:18888/job/ZOO-PR-Scala-Spark-2.2-k216/175/

File: pyzoo/zoo/pipeline/api/keras2/layers/pooling.py [hunk 10]
```diff
@@ -155,7 +155,7 @@ class GlobalAveragePooling2D(ZooKeras2Layer):
     input_shape (a shape tuple, does not include the batch dimension).
 
     # Arguments
-    data_format: Format of input data. Channel First .
+    data_format: Format of input data. Either channel_first  or channel_last.
```
Comment (hkvision) [inline pyzoo/zoo/pipeline/api/keras2/layers/pooling.py]: channels_first... channels_last

File: pyzoo/zoo/pipeline/api/keras2/layers/pooling.py [hunk 11]
```diff
@@ -121,3 +121,54 @@ def __init__(self,
...
+class GlobalAveragePooling2D(ZooKeras2Layer):
+    """
+    Applies global average pooling operation for spatial data.
+    The input of this layer should be 4D.
+
+    When you use this layer as the first layer of a model, you need to provide the argument
+    input_shape (a shape tuple, does not include the batch dimension).
+
+    # Arguments
+    data_format: Format of input data. Either channel_first  or channel_last.
```
Comment (hkvision) [inline pyzoo/zoo/pipeline/api/keras2/layers/pooling.py]: channels_first ... channels_last
